### PR TITLE
adminer-pematon: init at 4.12

### LINKED ISSUE #353454: Update request: adminer 4.8.1 → 4.11 (pematon fork)
- Package name: adminer
- Latest released version: 4.8.1
- Current version on the unstable channel: 4.8.1
- Current version on the stable/release channel: 4.8.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

It's been 3 years since the last activity in the [original repo](https://github.com/vrana/adminer/commits/master/).
It's safe to say that the project has been abandoned by the original maintainer, but few forks remain active.

One of them is [pematon@adminer](https://github.com/pematon/adminer), this particular fork is rapidly updating, bringing latest bug fixes to the package.
Another one is: https://github.com/adminerevo/adminerevo which is already packaged in nixpkgs (with plugin support)

Keeping a unmaintained package for long period of time may attract CVEs, although none has been assigned AFAIK.

Should we use this fork and update the `adminer` package?

**Notify maintainers**

@jtojnar @sstef - Adminer maintainer
@drupol - who approved the adminer evo
@Mic92 - adminer user

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by JohnRTitor**
Link: https://github.com/pematon/adminer

This is an active fork of adminer by pematon.
Seems to be in rapid development.

The index.php is required for plugins to work, and the upstream provides an example, but not a ".php" file in the src. We need to modify it for NixOS use anyway.

The index.php file is the same one used by `adminerevo`, I wasn't sure if I should reference a file out of the package's dir, so I copied it here.

Closes https://github.com/NixOS/nixpkgs/issues/353454

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
@ofborg build adminer-pematon

**Comment by JohnRTitor** [comment]
@ofborg build adminer-pematon

File: pkgs/by-name/ad/adminer-pematon/index.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace nixos {
```
Comment (drupol) [inline pkgs/by-name/ad/adminer-pematon/index.php:3]: Opinionated: I would modify this into something more functional, like:

```php
<?php

namespace nixos {
	function adminer_object(): object {
		require_once(__DIR__ . '/plugins/plugin.php');

		if (!file_exists(__DIR__ . '/plugins.json')) {
		    return new \AdminerPlugin();
		}

        $plugins = array_map(
            static function (string $name): ?object {
    			$plugin = sprintf('%s/plugins/%s.php', __DIR__, $name);

    			if (!is_readable($plugin)) {
    			    return null;
    			}

				require($plugin);

				preg_match_all('/(\w+)/', $name, $matches);

				return new sprintf('Adminer%s', implode('', array_map('ucfirst', $matches[1])));
            },
            json_decode(file_get_contents(sprintf('%s/plugins.json', __DIR__), true)
        );

		return new \AdminerPlugin(array_filter($plugins));
	}
}
```

**Review by drupol — APPROVED** [review]

File: pkgs/by-name/ad/adminer-pematon/index.php (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/ad/adminer-pematon/index.php:5]: Done, thanks, tested and it works.

One thing I might add though, do you know perhaps a way to use [adminer-theme](https://github.com/pematon/adminer-theme) here?

I am currently [removing the index.php](https://github.com/JohnRTitor/nix-conf/blob/57e2dd255de05a6f538d27352c9110693a83372e/pkgs/adminer-pematon-with-adminer-theme/default.nix#L26-L27) provided by this package, and adding an index.php with [modifications for adminer theme](https://github.com/JohnRTitor/nix-conf/blob/57e2dd255de05a6f538d27352c9110693a83372e/pkgs/adminer-pematon-with-adminer-theme/index.php#L25-L32).

**Pull request merged**